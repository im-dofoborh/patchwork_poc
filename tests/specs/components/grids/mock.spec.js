import { mount, createLocalVue } from '@vue/test-utils';
import MockGrid from '~/components/grids/Mock';

describe('MockGrid component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('mock-grid', MockGrid);
    wrapper = mount(MockGrid, {
      localVue,
      propsData: {
        width: 100,
      }
    })
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with alternative columns', () => {
    wrapper.setProps({
      columns: 6,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders fluid and fixed variant', () => {
    wrapper.setProps({
      fluidText: 'Foo',
      fixedText: 'Bar',
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
