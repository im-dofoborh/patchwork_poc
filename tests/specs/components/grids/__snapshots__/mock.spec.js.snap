// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockGrid component renders default variant 1`] = `
<div class="sg-mock-grid container" style="width: 100px;">
  <div class="sg-mock-grid__row row">
    <div class="col-12">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
  <div class="sg-mock-grid__row row">
    <div class="col-6">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-6">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
  <div class="sg-mock-grid__row row">
    <div class="col-4">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-4">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-4">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
  <div class="sg-mock-grid__row row">
    <div class="col-3">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-3">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-3">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-3">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
  <div class="sg-mock-grid__row row">
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-2">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
  <div class="sg-mock-grid__row row">
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
    <div class="col-1">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
</div>
`;

exports[`MockGrid component renders fluid and fixed variant 1`] = `
<div class="sg-mock-grid container" style="width: 100px;">
  <div class="sg-mock-grid__row row">
    <div class="sg-mock-grid__rail-container">
      <div class="sg-mock-grid__rail sg-mock-grid__rail--primary">
        <div class="sg-mock-grid__column">
          Foo
        </div>
      </div>
      <div class="sg-mock-grid__rail sg-mock-grid__rail--secondary">
        <div class="sg-mock-grid__column sg-mock-grid__column--rail">
          Bar
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MockGrid component renders with alternative columns 1`] = `
<div class="sg-mock-grid container" style="width: 100px;">
  <div class="sg-mock-grid__row row">
    <div class="col-6">
      <div class="sg-mock-grid__column"></div>
    </div>
  </div>
</div>
`;
